src/lib/transition_frontier_persistence/transition_frontier_persistence.ml

```diff
@@ -144,22 +138,30 @@ module Make (Inputs : Intf.Main_inputs) = struct
                         mutant ) )
           | None ->
               (* TODO: need to delete persistence once it get's back up *)
-              Deferred.return Diff_hash.empty )
+              Deferred.return Transition_frontier.Diff_hash.empty )
     in
     Deferred.unit
 
-  let directly_add_breadcrumb ~logger ~trust_system transition_frontier
-      transition_with_hash parent =
+  let directly_add_breadcrumb ~logger ~verifier ~trust_system
+      transition_frontier transition parent =
     let log_error () =
       Logger.fatal logger ~module_:__MODULE__ ~location:__LOC__
-        ~metadata:
-          [("hash", State_hash.to_yojson (With_hash.hash transition_with_hash))]
+        ~metadata:[("hash", State_hash.to_yojson (With_hash.hash transition))]
         "Failed to add breadcrumb into $hash"
     in
+    (* TODO: TMP HACK: our transition is already validated, so we "downgrade" it's validation *)
```

Comment:
reminder that this is here. what's TODO here? a variant of `Breadcrumb.build` that takes a fully validated transition?

Reply:
Removed the TODO and left the hack, but made an issue and comented with the number.
https://github.com/CodaProtocol/coda/issues/2486